zenoh/src/api/publisher.rs

```diff
@@ -518,6 +525,13 @@ pub struct MatchingStatus {
     pub(crate) matching: bool,
 }
 
+#[cfg(feature = "unstable")]
+#[derive(Debug, Copy, Clone, PartialEq)]
+pub(crate) enum MatchingStatusType {
+    Subscribers,
+    Queryables(bool),
+}
+
 #[zenoh_macros::unstable]
 impl MatchingStatus {
     /// Return true if there exist Subscribers matching the Publisher's key expression.
```

Comment:
Given the API change, then the doc should be updated as well.

@OlivierHecart we discussed in the past the possibility of having a matching status at session level. The API change proposed here is more generic, removing the need of having an explicit `matching_subscribers` in favour of `matching`. 

If I'm not mistaken, matching listeners can only be created from either a *publisher* and a *querier* as introduced by this PR. I'm fine with the proposed change but I want to hear your opinion on this.

Reply:
docs updated